examples/NewRelic.OpenTelemetry/ConsoleCore/Program.cs

```diff
@@ -34,26 +41,26 @@ static void Main(string[] args)
                     options.AuditLoggingEnabled = true;
                 }, loggerFactory)
                 .AddHttpClientInstrumentation()
-                .Build();
-
-            using (var activity = SampleActivitySource.StartActivity("SampleConsoleCoreAppSpan"))
+                .Build())
             {
-                Console.WriteLine("Creating root of trace.");
+                using (var activity = SampleActivitySource.StartActivity("SampleConsoleCoreAppSpan"))
+                {
+                    Console.WriteLine("Creating root of trace.");
 
-                var message = "Hello, OpenTelemetry with New Relic!";
-                activity?.SetTag("aNumber", 42);
-                activity?.SetTag("message", message);
+                    var message = "Hello, OpenTelemetry with New Relic!";
+                    activity?.SetTag("aNumber", 42);
+                    activity?.SetTag("message", message);
 
-                Console.WriteLine("\nMaking an external HTTP request which will be added as a child span of the trace.");
+                    Console.WriteLine("\nMaking an external HTTP request which will be added as a child span of the trace.");
 
-                var httpClient = new HttpClient();
-                var request = httpClient.GetAsync("https://www.newrelic.com");
+                    var httpClient = new HttpClient();
+                    var request = httpClient.GetAsync("https://www.newrelic.com");
 
-                Console.WriteLine($"Web request result: {request.Result.StatusCode}");
+                    Console.WriteLine($"Web request result: {request.Result.StatusCode}");
+                }
```

Comment:
You could put this in a separate method and then use `Task.Run` to do this in a `while(true)`. Then use a`Console.ReadLine` on the main thread to stop the app. Something like:

```csharp
static void Main()
{
    ...

   Task.Run(GenerateSpans);
   Console.WriteLine("Spans are being generated and exported to New Relic. Press enter to stop.");
   Console.ReadLine();
}

private void GenerateSpans()
{
    while (true)
    {
          using var activity = SampleActivitySource.StartActivity("SampleConsoleCoreAppSpan");

          ...
    }
}
```

Reply:
That's a good idea.  I've made the change.